rpcperms/middleware_handler.go

```diff
@@ -263,24 +263,41 @@ func (h *MiddlewareHandler) sendInterceptRequests(errChan chan error,
 					break
 				}
 
-				// For intercepted messages we also allow the
-				// content itself to be overwritten.
-				if t.ReplaceResponse {
-					response.replace = true
-					protoMsg, err := parseProto(
-						requestInfo.request.ProtoTypeName,
-						t.ReplacementSerialized,
+				// If there's nothing to replace, we're done,
+				// this request was just accepted.
+				if !t.ReplaceResponse {
+					break
+				}
+
+				// We are replacing the response, the question
+				// now just is: was it an error or a proper
+				// proto message?
+				response.replace = true
+				if requestInfo.request.IsError {
+					response.replacement = errors.New(
+						string(t.ReplacementSerialized),
```

Comment:
Don't consider this blocking, but wouldn't this _strip_ the error of any error code information? So if we returned a "nice" `status` gRPC error that one of the args was wrong, the user would only get the string and wouldn't be able to check for the error code as normal. 

Looking at the internals, I see that [`status.FromError`](https://github.com/grpc/grpc-go/blob/v1.47.0/status/status.go#L87) checks the error type for a special interface, and it's also possible to recover an error from an actual proto message as well. 

One thing I'm unsure of is: when we return the error on our gRPC handler function, is the error itself _always_ mapped into a proto? That would certainly make more sense, otherwise I don't see how they would make the pattern work across all the various language boundaries.

Reply:
Making this comment actually actionable, I think we can do the following:
  * If it's an error, try to use `FromError` to get the `Status` struct. 
  * Map that struct to the proto. 
  * Call `ErrorProto` later to map it back to the proper error proto 

In theory this would mean that the interceptor can make a "proper" proto error send that in, and have it be replaced, with the client-side being able to introspect into the rich proto error as normal. 

As mentioned above, I think it's fine to make this in to a follow up issue, since afaict we're able to do what we need to w/ this simpler version of error pass through.